src/models/bulkTransfer/bulkTransferAssociation.js

```diff
@@ -32,3 +33,28 @@ exports.create = async (bulkTransferAssociation) => {
     throw new Error(err.message)
   }
 }
+
+exports.update = async (transferId, bulkTransferId, bulkTransferAssociation) => {
+  try {
+    let record = {
+      bulkProcessingStateId: bulkTransferAssociation.bulkProcessingStateId,
+      lastProcessedDate: bulkTransferAssociation.lastProcessedDate || Time.getUTCString(new Date()),
+      errorCode: bulkTransferAssociation.errorCode,
+      errorDescription: bulkTransferAssociation.errorDescription
+    }
+    Object.keys(record).forEach(key => (record[key] === undefined || record[key] === null) && delete record[key])
```

Comment:
I've seen this a couple times now - perhaps it can be moved to a utils library, with a method name like `filterEmptyRecordInPlace` or something...

Reply:
We actually have it already as an util function that utilises lodash. I should probably revisit that list more often. Thanks.